src/search/include/search/create_controller.h

```diff
@@ -152,7 +156,16 @@ add_node_to_controller(
 		if (new_location) {
 			// To break circles in the search graph, only add the successor if it is actually a new
 			// location.
-			add_node_to_controller(successor.get(), K, controller);
+			add_node_to_controller(successor.get(),
+			                       controller_actions,
+			                       environment_actions,
+			                       K,
+			                       minimize_controller,
+			                       controller);
+		}
+		if (minimize_controller
+		    && controller_actions.find(timed_action.second) != std::end(controller_actions)) {
+			break;
```

Comment:
Short question here: This stops controller-creation if no controller action can be found, right? This might be a bit conservative (not necessarily wrong) in the sense that if we have only good environment-paths, this would still be a valid path, or not?

Reply:
No that should still work as before. But if we found a good controller action, we break, i.e., we ignore any (later) environment or controller actions.